numba/typing/npydecl.py

```diff
@@ -318,6 +318,29 @@ def generic(self, args, kws):
 
 builtin_global(numpy.ndindex, types.Function(NdIndex))
 
+@builtin
+class NdEmpty(AbstractTemplate):
+    key = numpy.empty
+
+    def generic(self, args, kws):
+        assert not kws
+        shape = args[0]
+        dtype = types.double
+        if len(args) >= 2:
+            npy_dtype = args[1]
+            dtype = from_dtype(numpy.dtype(npy_dtype.template.key))
```

Comment:
Is there a reason for the complication? Also, I'm wondering: does it work with a structured dtype?